From stable+bounces-81208-greg=kroah.com@vger.kernel.org Mon Oct  7 05:58:33 2024
From: Jonathan Gray <jsg@jsg.id.au>
Date: Mon,  7 Oct 2024 14:58:19 +1100
Subject: Revert "drm/amd/display: Skip Recompute DSC Params if no Stream on Link"
To: gregkh@linuxfoundation.org
Cc: stable@vger.kernel.org
Message-ID: <20241007035819.12246-1-jsg@jsg.id.au>

From: Jonathan Gray <jsg@jsg.id.au>

This reverts commit 6f9c39e8169384d2a5ca9bf323a0c1b81b3d0f3a.

duplicated a change made in 6.10.5
70275bb960c71d313254473d38c14e7101cee5ad

Cc: stable@vger.kernel.org # 6.10
Signed-off-by: Jonathan Gray <jsg@jsg.id.au>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/display/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -1267,9 +1267,6 @@ static bool is_dsc_need_re_compute(
 	if (new_stream_on_link_num == 0)
 		return false;
 
-	if (new_stream_on_link_num == 0)
-		return false;
-
 	/* check current_state if there stream on link but it is not in
 	 * new request state
 	 */
